src/bin/notification-proxy-client.rs

```diff
@@ -309,6 +314,15 @@ async fn client_server() {
                 .map
                 .remove(&sequence)
                 .expect("server violated the protocol")
+                // marmarek: is sending real ID back to the client a good idea? the specification
+                // mandates ID to be unique (until wraps) for the notification server uptime, but
+                // if the GUI domain's server is restarted but this one isn't, it may result in a
+                // duplicated ID returned (much) earlier.
+                // Maybe better send sequence number as the ID instead and keep mapping on the
+                // server side? and remove stale entries when notification is closed (you get a
+                // dbus signal) or when the actual notification server gets restart (name owner
+                // change signal).
+                // See also the comment about replaces_id, you likely need to keep such map anyway.
```

Comment:
Do you have any suggestions for a sequence number generator library?  Should I just borrow code from a Wayland library?

Reply:
Uhm, why do you need "a library" for a `sequence_number++`?!

Reply:
There are three cases where this isn’t trivial:

1. One needs to use 32-bit (as opposed to 64-bit) sequence numbers, and so needs to deal with wraparound.
2. One cares about performance enough to want to use an array instead of a hash table or binary tree.
3. One needs to reuse sequence numbers.

In both cases one needs a data structure that keeps track of the already-used sequence numbers.